### ISSUE TITLE
kata with cri-o and flannel: pod stuck in container create state

### ISSUE THREAD

**Issue by sdeoras** [kata-containers/runtime#2005]
# Description of problem

I am testing kata-containers using cri-o where I see pods are getting stuck in container create state. My setup is as follows:

- Ubuntu 18.04 nodes on GCP with nested virtualization enabled
- k8s 1.15.3
- Flannel for networking using IP forwarding on nodes

# Expected result

As a comparison I ran the same pods using `gVisor/runsc` and `runc` and both these cases worked fine on the same setup. `kata-runtime kata-check` and a simple `podman --runtime=/usr/bin/kata-runtime run --net=host ...` check are also working fine.

I expected kata-runtime based pod deployment to work similar to `runc` and `runsc`.

# Actual result

kubectl describe has following information:
```
Events:
  Type     Reason                  Age                  From                  Message
  ----     ------                  ----                 ----                  -------
  Normal   Scheduled               17m                  default-scheduler     Successfully assigned default/ping-pong-kata-server-deployment-65d878cd44-qcw8m to ubuntu-xpzc
  Warning  FailedCreatePodSandBox  16m                  kubelet, ubuntu-xpzc  Failed create pod sandbox: rpc error: code = Unknown desc = container create failed:
  Warning  FailedCreatePodSandBox  12m                  kubelet, ubuntu-xpzc  Failed create pod sandbox: rpc error: code = Unknown desc = container create failed: mkdir /var/lib/vc/sbs/f1a974521c23bbdb241e606500a545b15f7cc307d86b0fe18750b800f66694f9: cannot allocate memory
  Warning  FailedCreatePodSandBox  11m                  kubelet, ubuntu-xpzc  Failed create pod sandbox: rpc error: code = Unknown desc = container create failed: failed to create namespace: too many open files in system
  Warning  FailedCreatePodSandBox  10m (x2 over 12m)    kubelet, ubuntu-xpzc  Failed create pod sandbox: rpc error: code = Unknown desc = container create failed: Failed to check if grpc server is working: rpc error: code = Unavailable desc = transport is closing
  Warning  FailedCreatePodSandBox  2m4s (x40 over 17m)  kubelet, ubuntu-xpzc  Failed create pod sandbox: rpc error: code = Unknown desc = error reading container (probably exited) json message: EOF
```
Also attaching output of `kata-collect-data.sh` from the node where this pod was being scheduled.

[kata-collect-data.log](https://github.com/kata-containers/runtime/files/3547244/kata-collect-data.log)

**Comment by amshinde** [comment]
@sdeoras The kubectl describe output shows that you ran out of memory.
Is this a stress test that you are running with large number of pods.
Also, the line `failed to create namespace: too many open files in system` indicates that you ran out of number of allowed open files on the system. You should increase this limit with something like `sysctl -w fs.file-max=100000`

**Comment by sdeoras** [comment]
@amshinde Nope, this is not a stress test at all. It is a simple server-client communication check over gRPC. All it does is send ping-pong messages between client and server. For reference I have been evaluating the same with gVisor (and of course with runc) and both are fine.
I ran into a different issue with gVisor, but eventually things are working fine: https://github.com/google/gvisor/issues/807

I had also tried working with larger nodes (4 cores and more memory) but that did not help.
